src/metrics/kms-service.ts

```diff
@@ -29,11 +35,26 @@ export class KMSService {
       return result.Plaintext.toString("ascii");
     } catch (err) {
       if ((err as any).code === "MODULE_NOT_FOUND") {
-        const errorMsg = "optional dependency aws-sdk not installed. KMS key decryption will not work";
-        logError(errorMsg);
-        throw Error(errorMsg);
+        // Node 18
+        return this.decryptV3(buffer);
       }
       throw Error("Couldn't decrypt ciphertext");
     }
   }
+
+  // Node 18 or AWS SDK V3
+  public async decryptV3(buffer: Buffer): Promise<string> {
+    const { KMSClient, DecryptCommand } = require("@aws-sdk/client-kms");
+    const kmsClient = new KMSClient();
+    let result;
+    try {
+      const decryptCommand = new DecryptCommand({ CiphertextBlob: buffer });
+      result = await kmsClient.send(decryptCommand);
+    } catch {
+      const decryptCommand = new DecryptCommand({ CiphertextBlob: buffer, EncryptionContext: this.encryptionContext });
+      result = await kmsClient.send(decryptCommand);
```

Comment:
Is it guaranteed that this will never fail?

Reply:
no, but my above comment applies - this logic would raise through `decrypt` which would then continue.

I think I will modify the logic to always log, regardless of the error class though.